.github/workflows/opensearch-observability-test-and-build-workflow.yml

```diff
@@ -23,6 +23,9 @@ jobs:
       # need to switch to root so that github actions can install runner binary on container without permission issues.
       options: --user root
 
+    env:
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
```

Comment:
what is this? it shouldn't use node

Reply:
It is a change on github runners: 
```
On June 30th 2024, we will change the default from Node16 to Node20.
To opt out of this and continue using Node16 while it is still available in the runner, you can choose to set ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION=true
```

Reply:
but do we need it in this workflow? this workflow doesn't use node

Reply:
i understand the incompatibility, the question was where is it using node. i'm assuming it's the checkout action or github action internal impl

Reply:
Yes i think the checkout action uses node, did a quick search: https://github.com/search?q=repo%3Aactions%2Fcheckout%20node&type=code